modules/cache2k/src/main/scala/scalacache/cache2k/Cache2kCache.scala

```diff
@@ -0,0 +1,79 @@
+package scalacache.cache2k
+
+import java.time.temporal.ChronoUnit
+import java.time.{Clock, Instant}
+
+import org.cache2k.{Cache2kBuilder, Cache => CCache}
+import org.slf4j.LoggerFactory
+
+import scala.concurrent.duration.Duration
+import scala.language.higherKinds
+import scalacache.{AbstractCache, CacheConfig, Entry, Mode}
+
+/*
+ * Thin wrapper around cache2k.
+ *
+ * This cache implementation is synchronous.
+ */
+class Cache2kCache[V](underlying: CCache[String, Entry[V]])(implicit val config: CacheConfig,
+                                                            clock: Clock = Clock.systemUTC())
+    extends AbstractCache[V] {
+
+  override protected final val logger =
+    LoggerFactory.getLogger(getClass.getName)
+
+  def doGet[F[_]](key: String)(implicit mode: Mode[F]): F[Option[V]] = {
+    mode.M.delay {
+      val baseValue = underlying.peek(key)
+      val result = {
+        if (baseValue != null) {
+          val entry = baseValue.asInstanceOf[Entry[V]]
+          if (entry.isExpired) None else Some(entry.value)
+        } else None
+      }
+      logCacheHitOrMiss(key, result)
+      result
+    }
+  }
+
+  def doPut[F[_]](key: String, value: V, ttl: Option[Duration])(implicit mode: Mode[F]): F[Any] = {
+    mode.M.delay {
+      val entry = Entry(value, ttl.map(toExpiryTime))
+      underlying.put(key, entry)
```

Comment:
I've never used cache2k before, so I'm not very familiar with its API. But it looks like it supports setting the expiry time on a value. Maybe its better to use cache2k's built-in expiry functionality, instead of wrapping the value in an `Entry`?

https://cache2k.org/docs/1.0/apidocs/cache2k-api/org/cache2k/Cache.html#expireAt-K-long-